tools/pre_ci.py

```
@@ -0,0 +1,340 @@
+        STATUS_MAP (dict): Maps boolean or string outcomes to visual status labels.
+        _results (list): Internal log of (description, status) tuples for all steps.
+        is_ci (bool): True if executing within a remote CI environment.
+        min_ver (str): Minimum supported Python version for README updates.
+        max_ver (str): Maximum supported Python version for README updates.
+    """
+
+    STATUS_MAP = {
+        True: "✅ PASS",
+        False: "❌ FAIL",
+        "SKIPPED": "⏭️  SKIP",
+    }
```

Ruff (0.15.11)

[warning] Mutable default value for class attribute

(RUF012)